Makefile

```diff
@@ -32,37 +37,45 @@ help:
 	@echo "  eslint           Runs a code linter on the JavaScript code"
 	@echo "  dumpdb           Create a postgres database dump with timestamp used as file name"
 	@echo "  loaddb           Load a database dump into postgres, file name in DB_DUMP_FILE"
-	@echo "  build-frontend   Builds the frontend static files"
-	@echo "  build-frontend-w Watches the frontend static files and builds on change"
+	@echo "  build-tagadmin   Builds the tag_admin frontend static files"
+	@echo "  build-tagadmin-w Watches the tag_admin frontend static files and builds on change"
 	@echo "  sync-projects    Runs the synchronization task on all projects"
 	@echo "  requirements     Compiles all requirements files with pip-compile\n"
 
-.docker-build:
-	make build
-
-build:
-	cp ./docker/config/webapp.env.template ./docker/config/webapp.env
-	sed -i -e 's/#SITE_URL#/$(subst /,\/,${SITE_URL})/g' ./docker/config/webapp.env
+.frontend-build:
+	make build-frontend
+.webapp-build:
+	make build-webapp
 
+build: build-frontend build-webapp
+build-frontend: webapp-env
+	"${DC}" build frontend
+	touch .frontend-build
+build-webapp: webapp-env
 	"${DC}" build --build-arg USER_ID=$(USER_ID) --build-arg GROUP_ID=$(GROUP_ID) webapp
+	touch .webapp-build
 
-	touch .docker-build
+webapp-env:
+	cp ./docker/config/webapp.env.template ./docker/config/webapp.env
+	sed -i -e 's/#FRONTEND_URL#/$(subst /,\/,${FRONTEND_URL})/g;s/#SITE_URL#/$(subst /,\/,${SITE_URL})/g' ./docker/config/webapp.env
 
-setup: .docker-build
+setup: .webapp-build
 	"${DC}" run webapp //app/docker/set_up_webapp.sh
 
-run: .docker-build
-	"${DC}" run --rm --service-ports webapp
+run: .frontend-build .webapp-build
+	"${DC}" up --no-build frontend webapp
 
 clean:
-	rm .docker-build
+	rm -f .docker-build .frontend-build .webapp-build
+
+test: test-webapp test-frontend
 
-test:
+test-webapp:
 	"${DC}" run --rm webapp //app/docker/run_tests.sh
```

Comment:
This looks wrong. The `run_tests.sh` file is actually meant to run the full test suite, frontend and webapp/server included. It's basically what the CI runs, but in docker. 

If I understand things correctly, `test-webapp` should actually be an alias of `pytest` just like `test-frontend` is an alias of `jest`. And `test`, I think, should stay as it was. 

If that's not the case and I missed something, then the `run_tests.sh` file should probably be revisited. :-)

Reply:
At least at the moment, `run_tests.sh` also lints the code, i.e. it's not just an alias for `pytest`. Because it's run within the container, it can't after this change also run the tests for the translate frontend, as that's now in a different container. But it can still run the rest of its commands.

To keep the old behaviour of `make test`, we need these new targets, so that `test-webapp` and `test-frontend` can run all the tests that their corresponding containers offer.

Reply:
I also think `test-webapp` should only run the unit tests (i.e.  `pytest`) to be consistent with its `test-frontend` counterpart.

And `test` should run unit tests across the board, currently `test-webapp` and `test-frontend`.

I don't consider linting and formatting as tests, so I'd add a new command (i.e. `ci`) to run everything we run in CI.

Reply:
Made `make test-server` effetively an alias for `make pytest`, and added a new `make ci` that tests & lints the frontend & server, as is actually done in CI.